ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp

```diff
@@ -231,9 +231,19 @@ namespace enigma
 
     //NOTE: window_default() always centers the Window, GM8 only recenters the window when switching rooms
     //if the window size changes.
-    enigma_user::window_default(true);
-    // window sized by first room, can make visible now
-    enigma_user::window_set_visible(true);
+    if (!enigma::initGame) {
+      // GMS doesn't do this every room init; only game init
```

Comment:
But now you've made it so the window is not recentered when switching rooms, which is GM8 behavior. Perhaps compare it to the latest GMS master.

Reply:
GMStudio does not do what GM8 does in this regard. GM8 behavior in this case I feel YoYo made a good move when they dropped it. It's annoying to have the game window randomly move without the user's express permission. Plus, window_center() is broken on Xlib with multiple monitors, Josh wouldn't approve of GM8 behavior for that reason as well.

Reply:
I can confirm I can only reproduce it in GM8.1 but not the latest GMSv1.4, so your decision is fine. And to clarify, GM Classic only did it when the next room had a different size. If the two rooms were the same size, it would not move the window.